### ISSUE TITLE
Product operation along some axis

### ISSUE THREAD

**Issue by zldrobit** [pytorch/pytorch#15956]
## 🚀 Feature
An element-wise multiplication operation along axis, like numpy.prod or tf.reduce_prod

## Motivation
1. Since NumPy and TensorFlow have the corresponding operation, PyTorch should also have such op.
2. Sum be can applied along an axis, thus PyTorch may include this feature for completion.
3. In my use case, tensors are perceived as conditional/marginal probabilities, I want to multiply along an axis to calculate joint probabilities.

## Pitch
I don't know whether this feature is welcomed or it is decided PyTorch should not include it.

## Alternatives
1. Use torch.mul(A, B) or A * B in a loop.
2. Take the log values, add them up and take exponential of it.
3. Use torch.einsum

**Comment by t-vi** [comment]
Thank you for the suggestion. We do have `torch.prod`. Is there anything that you find lacking with it?

**Comment by zldrobit** [comment]
Thank you！At first I can not find this operation on google.

**Comment by t-vi** [comment]
As a rule of thumb, I'd always try if the numpy function is available. We do aspire to be largely compatible.

**Comment by Ashok93** [comment]
> Thank you！ 
Even I struggled to find the documentation of the same. Strange that it did not show up in Google.
